core/src/main/java/org/pentaho/di/connections/ConnectionManager.java

```diff
@@ -58,14 +61,45 @@ public class ConnectionManager {
   private static final Logger logger = LoggerFactory.getLogger( ConnectionManager.class );
 
   private List<LookupFilter> lookupFilters = new ArrayList<>();
+
+  @NonNull
   private Supplier<IMetaStore> metaStoreSupplier;
+
+  @NonNull
+  private final Bowl bowl;
+
   private ConcurrentHashMap<String, ConnectionProvider<? extends ConnectionDetails>> connectionProviders =
     new ConcurrentHashMap<>();
 
   private Map<String, List<String>> namesByConnectionProvider = new ConcurrentHashMap<>();
   private Map<String, ConnectionDetails> detailsByName = new ConcurrentHashMap<>();
   private boolean initialized;
 
+  public ConnectionManager() {
+    // Must throw a RuntimeException on metastore error to not break compatibility.
+    this( getMetastoreSupplierUnchecked( DefaultBowl.getInstance() ), DefaultBowl.getInstance() );
+  }
+
+  @VisibleForTesting ConnectionManager( @NonNull Supplier<IMetaStore> metaStoreSupplier, @NonNull Bowl bowl ) {
+    this.metaStoreSupplier = Objects.requireNonNull( metaStoreSupplier );
+    this.bowl = Objects.requireNonNull( bowl );
+  }
+
+  @NonNull
+  private static Supplier<IMetaStore> getMetastoreSupplier( @NonNull Bowl bowl ) throws MetaStoreException {
+    IMetaStore metastore = bowl.getMetastore();
+    return () -> metastore;
+  }
+
+  @NonNull
+  private static Supplier<IMetaStore> getMetastoreSupplierUnchecked( @NonNull Bowl bowl ) {
+    try {
+      return getMetastoreSupplier( bowl );
+    } catch ( MetaStoreException e ) {
+      throw new RuntimeException( e );
```

Comment:
ick. This is really not an appropriate use of unchecked exceptions. I really with the ConnectionManager API propagated exceptions in general. 
Returning null for errors is also bad. Blah. 

Fixing either of these would require a much larger change to the APIs, though. :(

Reply:
Yes... Only way would be to change the signature of the constructor, and then of `ConnectionManager#getInstance()`.

Anyway, it seems to me to be highly unlikely that this exception will happen in practice.
In the latest version, I created a custom runtime exception to get rid of the sonar cube reported code smell...

Reply:
Right. But, the supplier isn't used in the constructor, so the exception would be thrown later in each of the instance methods. 

Really irrelevant, though. We can't change those APIs.